pkg/streamformatter/streamformatter.go

```diff
@@ -170,3 +188,22 @@ func (sf *StderrFormatter) Write(buf []byte) (int, error) {
 	}
 	return len(buf), err
 }
+
+// AuxFormatter is a streamFormatter that writes aux progress messages
+type AuxFormatter struct {
+	io.Writer
+	*StreamFormatter
```

Comment:
I took a look and while I could just remove the `if sf.json` from `Emit` (since the else case is to output nothing) and I think I'd prefer to keep it since it makes the calling code/setup more consistent, e.g. it has:
```
	stdout := &streamformatter.StdoutFormatter{Writer: out, StreamFormatter: sf}
	stderr := &streamformatter.StderrFormatter{Writer: out, StreamFormatter: sf}
	...
	if versions.GreaterThanOrEqualTo(version, "1.30") {
		aux = &streamformatter.AuxFormatter{Writer: out, StreamFormatter: sf}
```
Where all of the output `Formatter`s share the same underlying `sf`. While I could pull the `if sf.json` up from `Emit` into the `versions` conditional above I don't really like the result.